src/openproduct/utils/oidc_drf_middleware.py

```diff
@@ -17,14 +20,24 @@ def authenticate(self, request):
 
         try:
             return super().authenticate(request)
+        except AssertionError as exc:
+            logger.exception(
+                "OIDC authentication failed due to assertion error: %s", exc
```

Comment:
Let's make sure it's an event and could you also add it to `logging.rst`? The exc_info should be automatically included because of the `.exception` call
```suggestion
                "oidc_authentication_failed"
```